### PR TITLE
Allow specifying dockerfile

### PR DESCRIPTION
**Pull request by banool**
This would allow users to specify a different Dockerfile than the default. The default is of course still `Dockerfile`. Feel free to change the name of the action back, I realise we might want it to keep the same name.

I have tested this as working on one of my own repos.

### CODE REVIEW

**Comment by janajri** [comment]
@VaultVulp any plan to merge this in? 

edit*: nvm, just used the fork.

**Comment by VaultVulp** [comment]
@janajri Thanks a lot for mentioning me! Looks like I've disabled some notification for this project, and didn't know that my github action got any attetion!

**Comment by VaultVulp** [comment]
Great idea, @banool ! Thanks a lot for your contribution, and sorry for late response!

**Comment by VaultVulp** [comment]
Everything is fine, @banool. Could you please resolve merge conflicts? I would do it myself, but it requires me to be able to push into your repo.

**Comment by banool** [comment]
I resolved the merge conflicts 😄 Thanks for looking at this!

**Comment by serucee** [comment]
Just a minor thing, but maybe change the readme from Dockerfile_server to docker-file-server or something to stay consistent with the naming

**Comment by janajri** [comment]
@banool I did attempt to use: banool/gp-docker-action but for some reason could not get the dockerfile to read from inside a directory. Any idea? 

*Looks like I was using the wrong version*

File: entrypoint.sh [hunk 1]
```diff
@@ -13,5 +14,5 @@ fi
 DOCKER_IMAGE_FULL_PATH=$(echo docker.pkg.github.com/${GITHUB_REPOSITORY}/${DOCKER_IMAGE_NAME}:${DOCKER_IMAGE_TAG} | tr '[:upper:]' '[:lower:]')
 
 docker login -u publisher -p ${DOCKER_TOKEN} docker.pkg.github.com
-docker build -t $DOCKER_IMAGE_FULL_PATH $BUILD_CONTEXT
+docker build -t $DOCKER_IMAGE_FULL_PATH $BUILD_CONTEXT -f $DOCKERFILE
```
Comment (VaultVulp) [inline entrypoint.sh]: Let's reorder arguments in a more consistent order

```suggestion
docker build -t $DOCKER_IMAGE_FULL_PATH -f $DOCKERFILE $BUILD_CONTEXT
```

File: action.yml [hunk 2]
```diff
@@ -15,6 +15,9 @@ inputs:
     description: 'Extract git-tag from repository'
     default: "false"
     required: false
+  dockerfile:
+    description: 'Dockerfile name'
+    default: "Dockerfile"
```
Comment (VaultVulp) [inline action.yml:20]: Let's make our new argument optional
```suggestion
    default: "Dockerfile"
    required: false
```

File: action.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: 'Build and publish Docker Image to GitHub Registry'
+name: 'Build and publish Docker Image to GitHub Registry with custom Dockerfile'
 description: 'GitHub Action to publish Docker Images to GitHub Registry'
```
Comment (VaultVulp) [inline action.yml]: I guess it's better to keep our Action's name as simple as possbile. But we indeed can update Action's desctription a bit.
```suggestion
name: 'Build and publish Docker Image to GitHub Registry'
description: 'GitHub Action to publish Docker Images to GitHub Registry. It's possible to specify custom Dockerfile and\or build context, if it's required.'
```

**Review by VaultVulp — CHANGES_REQUESTED** [review]

**Comment by banool** [comment]
I accepted the changes.

**Comment by VaultVulp** [comment]
> I accepted the changes.

Thanks a bunch! Have a great day, @banool !

**Pull request merged**